backend/tracim_backend/applications/share/lib.py

```diff
@@ -0,0 +1,172 @@
+from datetime import datetime
+from smtplib import SMTPException
+from smtplib import SMTPRecipientsRefused
+import typing
+import uuid
+
+from sqlalchemy.orm import Query
+from sqlalchemy.orm import Session
+
+from tracim_backend.applications.share.email_manager import ShareEmailManager
+from tracim_backend.applications.share.models import ContentShare
+from tracim_backend.applications.share.models import ContentShareType
+from tracim_backend.applications.share.models_in_context import ContentShareInContext
+from tracim_backend.config import CFG
+from tracim_backend.exceptions import NotificationSendingFailed
+from tracim_backend.exceptions import WrongSharePassword
+from tracim_backend.lib.mail_notifier.utils import SmtpConfiguration
+from tracim_backend.lib.utils.logger import logger
+from tracim_backend.lib.utils.utils import get_frontend_ui_base_url
+from tracim_backend.models.auth import User
+from tracim_backend.models.data import Content
+from tracim_backend.views import BASE_PUBLIC_API_V2
+
+FRONTEND_SHARED_CONTENT_LINK_PATTERN = "{frontend_ui_base_url}guest-download/{share_token}"
+PUBLIC_API_SHARED_CONTENT_LINK_PATTERN = (
+    "{api_base_url}{base_public_api}guest-download/{share_token}/{filename}"
+)
+
+
+class ShareLib(object):
+    def __init__(
+        self,
+        session: Session,
+        current_user: typing.Optional[User],
+        config: CFG,
+        show_disabled: bool = False,
+    ) -> None:
+        self._user = current_user
+        self._session = session
+        self._config = config
+        self.show_disabled = show_disabled
+
+    def base_query(self) -> Query:
+        base_query = self._session.query(ContentShare)
+        if not self.show_disabled:
+            base_query = base_query.filter(ContentShare.enabled == True)  # noqa: E712
+        return base_query
+
+    def share_content(
+        self,
+        content: Content,
+        emails: typing.List[str],
+        password: typing.Optional[str] = None,
+        do_notify=True,
+    ) -> typing.List[ContentShare]:
+        content_shares = []
+        created = datetime.now()
+        share_group_id = str(uuid.uuid4())
+        for email in emails:
+            content_share = ContentShare(
+                author=self._user,
+                content_id=content.content_id,
+                email=email,
+                share_token=str(uuid.uuid4()),
+                password=password,
+                type=ContentShareType.EMAIL,
+                created=created,
+                share_group_id=share_group_id,
+                enabled=True,
+            )
+            self.save(content_share)
+            content_shares.append(content_share)
+            self._session.flush()
+
+        if do_notify and self._config.EMAIL__NOTIFICATION__ACTIVATED:
```

Comment:
Maybe an error when `do_notify` is True and `self._config.EMAIL__NOTIFICATION__ACTIVATED` is False ?

Reply:
fixed, do_notify get  self._config.EMAIL__NOTIFICATION__ACTIVATED value.